src/drivers/README.md

```diff
@@ -1,4 +1,10 @@
 # Drivers
 
-### Misc Notes:
-- the `SerialDetection` class can get confused if one device is unplugged, and another one is plugged back in and allocated the same serial port as the first, all before the `getSerialStream()` is called again
+## Determining Variance for encoder_to_odometry node
+The `linear.x`, `linear.y`, and `angular.z` in the `Odometry` msg that we publish based on encoder data all have associated variances that are stored along the diagonal of the `pose.covariance` matrix (for full documentation, see [Odometry msg documentation](http://docs.ros.org/api/nav_msgs/html/msg/Odometry.html)). These are set as parameters on the `encoder_to_odometry` node. These values are determined experimentally, and are different for every robot. Here are the steps to determine them:
+
+1. Start the `encoder_to_odometry` node, and push the robot in a curve with a radius of at least 1 meters (the larger the better, just make sure you are pushing it in the exact same curve every time). Repeat this at least 10 times, re-starting the node each time, and recording the end `linear.x`, `linear.y`, `angular.z`, and `angular.w` values. (**Note:** The orientation in the `Odometry` msg is a quaternion. To get `theta` in radians, use an [online quaternion-radian converter](http://quaternions.online/) or use the math [from wikipedia](https://en.wikipedia.org/wiki/Conversion_between_quaternions_and_Euler_angles#Source_Code_2)).
```

Comment:
Euler angles are yaw, pitch, roll angles, its the fundamental set of angles to get into any orientation in space you want. In our case, is it correct to be looking at conversion from Euler Angles to Quaternions? Or do we want to be looking at something else?

Reply:
Basically there is a mixture of notation here. `linear.x,y` are used in quaternions (euclidean space), `angular.z` is a Euler angle (generally coming in the whole set of `angular.x,y,z`, or around an oriented vector), whilst `angular.w` is a vector angle used in the quaternion ROS data type to describe a rotation. 

What I'm saying is, are you sure you are recording `angular.z` AND `angular.w` at the same time? Because by Euler's rotation rule, `angular.w` will encompass `angular.z`.

Reply:
I'm not sure I totally understand what you're getting at here? I was just trying to match the instructions with what someone looking at the output of `rostopic echo` would see (ie. `linear.x`, `linear.y`, `angular.w`, `angular.z`)? Do you not need `angular.z` and `angular.w` to calculate the yaw of the robot?

Reply:
`angular.z` is yaw of the robot. `angular.w` is the combination of yaw, pitch and roll on a chosen axis. Which is generally your x,y,z vector in quaternions.

Reply:
Sorry, I still don't understand. I thought that you needed both the `z` and `w` components of a quaternion (which is what angular is) to get the yaw?

Reply:
Ok, that makes sense, but I'm still not exactly sure what you're getting at here? Just to make sure we're on the same page, `linear.x,y,z` describe where we are as a point in 3D space (and I don't believe is in any way related to our orientation), and `angular.x,y,z,w` describe our orientation in 3D space in the form of a quaternion?

Reply:
It is `angular x,y,z` or `angular.w` chose one. Not both. They represent the same thing. Standard quaternion should be `angular.w`. `angular x,y,z` is not quaternion notation.

Reply:
Ah, I think I understand. Here I'm just trying to follow ROS notation as much as possible to make it easy to follow (which does use `x,y,z,w` for quaternion representation, see http://docs.ros.org/api/geometry_msgs/html/msg/Quaternion.html).

Reply:
Ahh, ok, I think I understand now, changes made, please re-review.